capa/helpers.py

```diff
@@ -76,13 +77,39 @@ def load_json_from_path(json_path: Path):
         try:
             report_json = compressed_report.read()
         except gzip.BadGzipFile:
-            report = json.load(json_path.open(encoding="utf-8"))
+            report = json.decode(json_path.read_text(encoding="utf-8"))
         else:
-            report = json.loads(report_json)
+            report = json.decode(report_json)
     return report
 
 
+def load_jsonl_from_path(jsonl_path: Path) -> Iterator[Dict]:
+    with open(jsonl_path, "rb") as f:
+        for line in f:
+            try:
+                line_s = line.strip().decode()
+                obj = json.decode(line_s)
+                yield obj
+            except:
+                # ignore erroneous lines
+                continue
+
+
+def load_one_jsonl_from_path(jsonl_path: Path) -> str:
+    # this loads one json line to avoid the overhead of loading the entire file
+    with open(jsonl_path, "rb") as f:
+        line = next(iter(f))
+        line = json.decode(line.decode(errors="ignore"))
+    return line
+
+
 def get_format_from_report(sample: Path) -> str:
+    if sample.name.endswith(".log"):
+        line = load_one_jsonl_from_path(sample)
+        if "Plugin" in line:
+            return FORMAT_DRAKVUF
```

Comment:
Hmm I suppose not necessarily but I assumed that since we're using msgspec and since that routine (load_one_jsonl_from_path) is loading only a single jsonl line then this wouldn't be inefficient.